src/main/java/org/ibp/api/java/impl/middleware/study/StudyEntryServiceImpl.java

```diff
@@ -104,7 +104,7 @@ public StudyEntryDto replaceStudyEntry(final Integer studyId, final Integer entr
 			germplasmList.getListData().stream().map(l -> mapper.map(l, StudyEntryDto.class)).collect(Collectors.toList());
 
 		final Map<Integer, GermplasmListData> germplasmListDataMap =
-			germplasmList.getListData().stream().collect(Collectors.toMap(GermplasmListData::getGermplasmId, g -> g));
```

Comment:
I'm not sure if we should be using GID as map key here given we can have duplicate GIDs within a list. I think we should be using entry number (entry ID) as key. 
What do you think @mderamos ?

Reply:
Hi @darla-leafnode I agree. We should use the  entry number. Sorry, I missed it. 
Hi @gelli-leafnode , please revise the code to  use entry number as key. Thanks!